### DISCUSSION TITLE
Feature Contribution Help!

### DISCUSSION THREAD

**Discussion by salahineo** [ActivityWatch/activitywatch#672] [Q&A] [answered]
Hi Everyone,

I’m Mohamed Salah. Frontend Developer.
I really love this application with its amazing features, but unfortunately, I didn’t found the dark mode feature ready yet.

So I decided to help (by the experience of my job) and make a CSS file that overrides the web app theme to be dark,
Then I've added the functionality to let users change the theme from settings and save this change to localStorage. 

Here are screenshots for settings select menu which users can change application theme from it:
![image](https://user-images.githubusercontent.com/49915551/135748523-5174ecef-6c41-4789-aa36-aebbc1230edb.png)
![image](https://user-images.githubusercontent.com/49915551/135748554-e4e87938-7439-4b17-9727-abe1f0c6fbd6.png)

Here are screenshots for the dark mode view:
![image](https://user-images.githubusercontent.com/49915551/135748591-dbf206c9-46c8-4e88-8128-38cb5da2e01b.png)
![image](https://user-images.githubusercontent.com/49915551/135748614-4d68d74e-fa8d-40b8-abdb-105e0be8101c.png)

Now I've only one issue,
That is my first time to contribute to an open source project, so I don't know the required steps for this contribution.

Step I followed to implement this feature:
1. Clone the [Web-UI](https://github.com/ActivityWatch/aw-webui) repository
2. Create a new branch called `dev/dark-theme`
3. Implement feature by adding new files and writing some codes
4. Commit my changes to this local branch 
5. [Issue Here] There is a permission error while publishing my local branch or pushing changes to remote repo

I don't know if this is the correct approach of contributing to this project, but it is the approach I used when working on my company projects.

So can anyone help me how to push my changes to this repo and create a new release with this feature.
Thanks.

**Comment by ErikBjare** [comment, marked as answer]
Hi Mohamed, changes look great!

To contribute a PR, you first need to fork the repo (using the leftmost button in the top-right corner): 
![image](https://user-images.githubusercontent.com/1405370/136044636-6d8efdda-9d08-4649-9e1e-565be31eb7dc.png)

Once you've forked the repo, you can change the remote of your branch using `git remote set-url <url to your forked repo>`, and then push as normal. Once you've pushed to your own branch, you can open a PR with the changes!

**Reply by salahineo** [reply]
Thank you,
I've opened a PR for this feature now.

**Reply by peepo5** [reply]
I see you say you opened a PR but I searched by your account name and nothing comes up?

**Reply by salahineo** [reply]
@peepopoggers 
Here is the pull request URL:
https://github.com/ActivityWatch/aw-webui/pull/307

**Comment by JFMugen** [comment]
That looks awesome. Just wanted to say to @salahineo, you could/can use Dark Reader extension as well. Not just for AW, for pretty much everything. (only thing it didn't work was my modem interface so far)

If you are like me who use EVERYTHING with dark mode, then it is a godsend ^^

**Reply by salahineo** [reply]
Thanks, it is really helpful 👍
